### PR TITLE
signing: add new ECDSA based signature messages

### PR DESCRIPTION
**Pull request by silverjam**
# Description

Adds new messages:
- ECDSA signature message
- ECDSA certificate message for IC and CC
- New "certificate chain" message (i.e. allow list)

@swift-nav/devinfra

# API compatibility

*Does this change introduce a API compatibility risk?*

No, just adds some new messages and deprecates some existing ones.

# JIRA Reference

- https://swift-nav.atlassian.net/browse/GV2-191
- https://swift-nav.atlassian.net/browse/GV2-192
- https://swift-nav.atlassian.net/browse/GV2-193

### CODE REVIEW

**Comment by silverjam** [comment]
Most of the pertinent changes are here: https://github.com/swift-nav/libsbp/pull/1302/files#diff-0daef5fdde7e96323ae084523dcf809462bda38b3960216da50007f9cf805c7f

**Comment by Blast545** [comment]
@silverjam Do we have to deprecate the ED25519 messages? I was talking with RK that in some time (not so sure future) we might want to support having different algorithms. 

I was wondering if it was better to skip deprecating them for now or to just deprecating them and adding them again if we ever decide to go that way.

File: c/include/libsbp/v4/signing/MSG_ECDSA_SIGNATURE.h [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+ *
+ * SBP_MSG_ECDSA_SIGNATURE
+ *
+ *****************************************************************************/
+/** An ECDSA signature
+ *
+ * An ECDSA-256 signature using SHA-256 as the message digest algorithm.
+ */
+typedef struct {
+  u8 flags;
```
Comment (Blast545) [inline c/include/libsbp/v4/signing/MSG_ECDSA_SIGNATURE.h:46]: Should we document here how the flags byte it's going to be used?

**Review by Blast545 — APPROVED** [review]

File: c/include/libsbp/legacy/signing.h [hunk 2]
```diff
@@ -25,16 +25,94 @@
...
+                                        certificate, intermediate
+                                        certificate and corrections
+                                        certificate. */
+  u8 signature[64];                /**< An ECDSA signature (created by the root
+                                        certificate) over the concatenation of
+                                        the SBP payload bytes preceding this
+                                        field (that is: the concatenation of
+                                        `root_certificate`,
+                                        `intermediate_certificate`, and
+                                        `corrections_certificate`).  This
```
[the remark this answers is not in the document]
Comment (silverjam) [reply c/include/libsbp/legacy/signing.h:77]: Yes

File: c/include/libsbp/legacy/signing.h [hunk 3]
```diff
@@ -25,16 +25,94 @@
...
+                                        certificate chain "allow list" can also
```
[the remark this answers is not in the document]
Comment (silverjam) [reply c/include/libsbp/legacy/signing.h:78]: :+1:

File: c/include/libsbp/v4/signing/MSG_ECDSA_SIGNATURE.h (hunk 1 above)
Comment (silverjam) [reply c/include/libsbp/v4/signing/MSG_ECDSA_SIGNATURE.h:46]: Fixed.

**Comment by silverjam** [comment]
> @silverjam Do we have to deprecate the ED25519 messages? I was talking with RK that in some time (not so sure future) we might want to support having different algorithms.

I think we should deprecate for now since we're already changing the format of the signature messages (in the `certificate_id` for example).

**Review by notoriaga — APPROVED** [review]

**Comment by silverjam** [comment]
Merging, only failure is a transient benchmark failure

**Pull request merged**